### DISCUSSION TITLE
Request Draining Support for Seamless Rolling Deployments

### DISCUSSION THREAD

**Discussion by 1finedev** [coollabsio/coolify#4692] [Improvement Requests]
I’ve noticed an issue during rolling deployments, once a new container is live, the old container is immediately deleted, even if it is still handling active requests. This abrupt switch causes serious issues, such as incomplete operations.

While i understand this helps the vast majority conserve resources, would it be possible to consider adding an option to allow the old container to drain all active requests for a user defined timer before being replaced? This approach would ensure smoother transitions, even if it means running two containers momentarily. It would greatly enhance reliability for services with long-running requests or critical database operations.

**Comment by Abdussalam-Mujeeb-ur-rahman** [comment]
Exactly!
I second this

**Reply by risenphoenix1234** [reply]
Valid 👍 💯

**Comment by OlaoluwaVincent** [comment]
Having a feature that keeps the container active for a while to drain out active request would be really great.

**Comment by Mbazu-Daniel** [comment]
Yea this is great, I know they are helping in cleaning the system but we should have the option to manually delete the old container if we are satisfied that the new container has what we want

**Comment by 1finedev** [comment]
Community turning up for this is cool, real need for us all

**Comment by d-a-ve** [comment]
Having this feature will be really good. An application I have completely removes the old container once the new container is ready and this causes some of our pages to have no data but after like 5mins or so, the page is right and working as expected.

Yes, not every app will need this but there are some apps that having this feature would make life way easier for the devs and users.

**Comment by jcardonne** [comment]
+1
